### PR TITLE
Clarify in `--help` that `cargo test --all-targets` excludes doctests

### PR DESCRIPTION
**Pull request by obi1kenobi**
Per my proposal here: https://github.com/rust-lang/cargo/issues/6669#issuecomment-1658593167

I tried to keep the edit minimalistic to match the surrounding style.

If the maintainers are amenable to it, I think it could also be useful to do one or more of:
- Offer concrete guidance on what to do to run actually-all tests (`--all-targets` then separately `--doc`).
- Link to the issue at: https://github.com/rust-lang/cargo/issues/6669
- Mention that `cargo test` without `--all-targets` runs doctests by default, which seems not immediately obvious.

I'd be happy to attempt to add any of the above that the maintainers feel would be a good fit here.

### CODE REVIEW

**Comment by ehuss** [comment]
I left a comment over at https://github.com/rust-lang/cargo/issues/6669#issuecomment-1658829220. This isn't a review on this PR (I'll leave that up to epage), but just a passing comment that saying "except doctests" might imply that a doctest is a target (which it isn't). I would probably word this like `(does not include doctests)` or something like that, but I am splitting hairs and the exact wording isn't important to me, just a suggestion if you like it better.

I do agree it is helpful to clarify, so thank you for opening this!

**Comment by epage** [comment]
> If the maintainers are amenable to it, I think it could also be useful to do one or more of:

I think at least clarifying the webpage / man page / `cargo help test` in the areas [I highlighted](https://github.com/rust-lang/cargo/issues/6669#issuecomment-1658861048) would be good.  We could fix the current miscommunication and handle this at the same time.

As for the ``-h` / --help`, we generally keep that brief (and I think `cargo add` is the only command with separate `-h` and `--help` on top of the man page)

File: src/bin/cargo/commands/test.rs [hunk 1]
```diff
@@ -34,7 +34,7 @@ pub fn cli() -> Command {
             "Test all tests",
             "Test only the specified bench target",
             "Test all benches",
-            "Test all targets",
+            "Test all targets except doctests",
```
Comment (epage) [inline src/bin/cargo/commands/test.rs:37]: > >  I would probably word this like (does not include doctests) or something like that, but I am splitting hairs and the exact wording isn't important to me, just a suggestion if you like it better.

I like the suggestion for it being paranthetical

I think I also prefer wording besides "except" makes it sound like we are actively excluding it
- `does not include` is ok
- `precludes` might also work

Comment (obi1kenobi) [reply src/bin/cargo/commands/test.rs:37]: I can switch to `does not include` with a parenthetical.

Though, one thing that phrasing doesn't address or even hint at is that `cargo test` without this flag _does_ run doctests. From that perspective, adding this flag does have the effect of actively excluding doctests, even though I understand the difference in intent there.

**Comment by obi1kenobi** [comment]
> I think at least clarifying the webpage / man page / cargo help test in the areas https://github.com/rust-lang/cargo/issues/6669#issuecomment-1658861048 would be good. We could fix the current miscommunication and handle this at the same time.

I don't feel like I understand the finer points of targets well enough to tackle the "docs text implies doctests are a target" issue in the links you shared, but I did open another PR to add a clarification for the `--all-targets` flag on that page: https://github.com/rust-lang/cargo/pull/12423

**Comment by obi1kenobi** [comment]
I'm running into some linking-related problem where even though I have `pkg-config` and OpenSSL installed as recommended in the README, cargo is still failing to build.

I tried to edit the snapshot test by hand so as not to block this PR on the issues with my local setup.

**Comment by weihanglo** [comment]
> I'm running into some linking-related problem where even though I have `pkg-config` and OpenSSL installed as recommended in the README, cargo is still failing to build.

Off topic but I am interested in this. If you are okay with it please share the detail on zulip or in a new issue :)

**Comment by epage** [comment]
Thanks!

@bors r+

**Pull request merged**